src/Symfony/Component/Messenger/Middleware/HandleMessageMiddleware.php

```diff
@@ -66,6 +78,21 @@ public function handle(Envelope $envelope, StackInterface $stack): Envelope
             $this->logger->info('No handler for message "{class}"', $context);
         }
 
+        if (\count($exceptions)) {
+            throw new HandlerFailedException($envelope, ...$exceptions);
+        }
```

Comment:
Just thinking out loud: one practical implication is that, if someone listens on the new `WorkerMessageFailedEvent` event, they will always (well, not technically "always", but pretty much always) receive *this* exception instead of the actual exception. Then they'll need to loop over `getNestedExceptions()` if they want to look for a specific exception. I think that's ok, just stating that.

Reply:
Yes, exactly. I was thinking whether it would make sense or not to throw the original exception if there is only one but it would mean you need to catch your own exception plus `HandlerFailedException`. So better always throwing it.